clang/test/SemaSYCL/intel-fpga-local-ast.cpp

```diff
@@ -0,0 +1,506 @@
+
+  // Check duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_private_copies
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelPrivateCopiesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  //CHECK-NOT:  SYCLIntelPrivateCopiesAttr
+  [[intel::private_copies(12)]]
+  [[intel::private_copies(12)]] int var_private_copies;
+
+  // Checking of duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_forcep2d
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}1
+  //CHECK-NEXT: IntegerLiteral{{.*}}1{{$}}
+  //CHECK-NOT:  SYCLIntelForcePow2DepthAttr
+  [[intel::force_pow2_depth(1)]]
+  [[intel::force_pow2_depth(1)]] int var_forcep2d;
+
+  // Test for Intel FPGA bankwidth memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_bankwidth 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 16
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 16
+  //CHECK-NOT:  SYCLIntelBankWidthAttr
+  [[intel::bankwidth(16)]]
+  [[intel::bankwidth(16)]] int var_bankwidth;
+
+  // Test for Intel FPGA numbanks memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_numbanks 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //CHECK-NOT:  SYCLIntelNumBanksAttr
+  [[intel::numbanks(8)]]
+  [[intel::numbanks(8)]] int var_numbanks;
+
+  // Checking of different argument values.
+  //CHECK: VarDecl{{.*}}bw_bw 'unsigned int[64]'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //CHECK-NOT:  SYCLIntelBankWidthAttr
+  [[intel::bankwidth(8)]]
+  [[intel::bankwidth(16)]] unsigned int bw_bw[64];
+
+  //CHECK: VarDecl{{.*}}nb_nb 'unsigned int[64]'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //CHECK-NOT:  SYCLIntelNumBanksAttr
+  [[intel::numbanks(8)]]
+  [[intel::numbanks(16)]] unsigned int nb_nb[64];
+
+  //FIXME: Last one is applied and other is ignored.
+  //CHECK: VarDecl{{.*}}mrg_mrg
+  //CHECK: SYCLIntelMergeAttr{{.*}}"mrg5" "width"{{$}}
+  [[intel::merge("mrg4", "depth")]]
+  [[intel::merge("mrg5", "width")]] unsigned int mrg_mrg[4];
+
+  // FIXME: Duplicate attribute is ignored.
```

Comment:
Same here. The current FIXME is confusing because it sounds like the issue is that we're ignoring attribute

Reply:
Done